test/Array/pop4.js

```diff
@@ -0,0 +1,32 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (c) 2022 ChakraCore Project Contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+const array = [1, 2, 3];
+
+const testRuns = 100_000;
+let counter = 0;
+
+let proto = [];
+Object.defineProperty(proto, 3, {
+    get: function () {
+        counter++;
+        return 42;
+    }
+});
+array.__proto__ = proto;
+
+function hotFunction() {
+    array.pop();
+}
+
+for (let i = 0; i < testRuns; i++) {
+    array.length = 4;
+    hotFunction();
+}
+
+if (counter < testRuns) {
+    throw new Error(`Expected ${testRuns} calls to getter, got ${counter}`);
+}
+print("pass");
```

Comment:
One of our style checks enforces a blank line at the end of every file - you'll need to add that here.